backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/ObservationHandler.java

```diff
@@ -228,34 +232,70 @@ public ObservationHandler(Executor executor, FutureJdbi jdbi) {
   public InternalFuture<Void> deleteObservations(String runId, Optional<List<String>> maybeKeys) {
     return jdbi.useHandle(
         handle -> {
-          // Delete from keyvalue
-          var sql =
-              "delete from keyvalue where id in "
-                  + "(select keyvaluemapping_id from observation where entity_name=:entity_name and field_type=:field_type and experiment_run_id=:run_id)";
+          handle.useTransaction(
+              TransactionIsolationLevel.SERIALIZABLE,
+              handle1 -> {
+                // Delete from keyvalue
+                var fetchQueryString = "select o.id, o.keyvaluemapping_id from observation as o ";
+                if (maybeKeys.isPresent()) {
+                  fetchQueryString += " INNER JOIN keyvalue as kv ON o.keyvaluemapping_id = kv.id ";
+                }
+                fetchQueryString +=
+                    " WHERE o.experiment_run_id = :run_id " + " AND o.entity_name = :entityName";
 
-          if (maybeKeys.isPresent()) {
-            sql += " and kv_key in (<keys>)";
-          }
+                if (maybeKeys.isPresent()) {
+                  fetchQueryString += " AND kv.kv_key IN (<keys>)";
+                }
+                var query =
+                    handle
+                        .createQuery(fetchQueryString)
+                        .bind("run_id", runId)
```

Comment:
This might require quotes for SQLserver compatibility

Reply:
We are removing quotes for SQLserver compatibility so no need to add it